march_hardware/src/EtherCAT/EthercatSDO.cpp

```diff
@@ -26,7 +26,19 @@ int sdo_bit8(uint16_t slave, uint16_t index, uint8_t sub, uint8_t value)
   return received_working_counter;
 }
 
-int sdo_bit16(uint16_t slave, uint16_t index, uint8_t sub, uint16_t value)
+int sdo_bit8_read(uint16_t slave, uint16_t index, uint8_t sub, int* val_size, uint8_t* value)
+{
+  ROS_DEBUG("sdo_bit8: slaveIndex %i, reg 0x%X, subindex %i, value 0x%X", slave, index, sub, value);
+  int received_working_counter = ec_SDOread(slave, index, sub, FALSE, val_size, value, EC_TIMEOUTRXM);
+  if (received_working_counter == 0)
+  {
+    ROS_FATAL("(sdo_bit8) Error occurred when writing, reg 0x%X, sub-index %i, data: 0x%X to slave %i", index, sub,
```

Comment:
```suggestion
    ROS_FATAL("(sdo_bit8) Error occurred when reading, reg 0x%X, sub-index %i, data: 0x%X from slave %i", index, sub,
```
Maybe de printing the data is also not so useful here :innocent: